app/src/utils/experimentsHelpers.ts

```diff
@@ -7,3 +7,25 @@ export const getExperimentNameFromId = (
   allExperiments.find(
     (anExperiment: Experiment): boolean => anExperiment.id === anId
   )?.name ?? '';
+
+export const insertExperimentAt = (
+  sourceList: Experiment[],
+  anExperiment: Experiment,
+  targetIndex: number
+): Experiment[] =>
+  sourceList.length === 0
+    ? [anExperiment]
+    : [
+        ...sourceList.splice(0, targetIndex),
```

Comment:
Thats the error that you cannot insert in the middle. It has to be slice

Reply:
Just merged your fix from develop into this branch - fixed now :)